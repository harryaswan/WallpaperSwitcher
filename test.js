
const application = require("./lib/app.js");
const os = require("./lib/os");
const path = require('path');
global.appRoot = path.resolve(__dirname);

var inArg = process.argv[2];
var configFile = inArg === 'debug' ? "/debug.json" : "/config.json";

os.loadFile(global.appRoot + configFile)
.then(function (data) {
    var config = JSON.parse(data);
    console.log('config loaded');
    var app = new application(config);

    os.checkFolders();

    if (inArg && inArg !== "debug") {
        app.handleURL(inArg)
    } else {
        app.go(1);
    }

})
.catch(function (err) {
    console.log("Could not load config", err);
})



// app.downloadImage("http://i.imgur.com/TAK4bet.jpg", '.', 'test.jpg', function (err, path) {
//     if(err) {
//         console.error(err);
//     } else {
//         console.log('Path', path);
//     }
// });

// var spoofData = {
//     data: {
//         children: [
//             {
//                 data: {
//                     domain: "imgur.com",
//                     url: "http://imgur.com/a/7tTEJ"
//                 }
//             }
//         ]
//     }
// }
//
// app.handleData(spoofData);
//
// console.log("curr dir", __dirname);



// app.imgurDownload("http://imgur.com/a/7tTEJ")
// .then(function (path) {
//     console.log("GOT PATH:", path);
// })
// .catch(function (err) {
//     console.error(err);
// })
