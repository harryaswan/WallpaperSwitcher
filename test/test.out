Running run number 1...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: http://imgur.com/gallery/WLnSe2J
Getting data: { url: 'https://api.imgur.com/3/image/WLnSe2J',
  headers: { Authorization: 'Client-ID 905f8623e29ae6a' } }
WLnSe2J.jpg 7
Getting data: { url: 'http://i.imgur.com/WLnSe2J.jpg', encoding: 'binary' }
set imgrwal
set wallpaper
Running run number 2...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: http://imgur.com/a/JEr9v
Getting data: { url: 'https://api.imgur.com/3/image/JEr9v',
  headers: { Authorization: 'Client-ID 905f8623e29ae6a' } }
Getting data: { url: 'https://api.imgur.com/3/album/JEr9v/images',
  headers: { Authorization: 'Client-ID 905f8623e29ae6a' } }
got data
jlOtgFU.jpg 7
Getting data: { url: 'http://i.imgur.com/jlOtgFU.jpg', encoding: 'binary' }
set imgrwal
set wallpaper
Running run number 3...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://i.redd.it/1u1fcteoepmx.jpg
1u1fcteoepmx.jpg 12
Getting data: { url: 'https://i.redd.it/1u1fcteoepmx.jpg',
  encoding: 'binary' }
set wallpaper
Running run number 4...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://i.redd.it/bhemqwzuupmx.png
bhemqwzuupmx.png 12
Getting data: { url: 'https://i.redd.it/bhemqwzuupmx.png',
  encoding: 'binary' }
set wallpaper
Running run number 5...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://i.redd.it/psz1nb9etdmx.jpg
psz1nb9etdmx.jpg 12
Getting data: { url: 'https://i.redd.it/psz1nb9etdmx.jpg',
  encoding: 'binary' }
set wallpaper
Running run number 6...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: http://i.imgur.com/dLoIkF9.png
dLoIkF9.png 7
Getting data: { url: 'http://i.imgur.com/dLoIkF9.png', encoding: 'binary' }
set wallpaper
Running run number 7...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://i.redd.it/pd8cc1mhgimx.jpg
pd8cc1mhgimx.jpg 12
Getting data: { url: 'https://i.redd.it/pd8cc1mhgimx.jpg',
  encoding: 'binary' }
set wallpaper
Running run number 8...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://cdnb3.artstation.com/p/assets/images/images/000/744/299/large/shen-fei-arrival.jpg?1443930283
shen-fei-arrival.jpg?1443930283 16
Getting data: { url: 'https://cdnb3.artstation.com/p/assets/images/images/000/744/299/large/shen-fei-arrival.jpg?1443930283',
  encoding: 'binary' }
set wallpaper
Running run number 9...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: http://imgur.com/a/PAERb
Getting data: { url: 'https://api.imgur.com/3/image/PAERb',
  headers: { Authorization: 'Client-ID 905f8623e29ae6a' } }
Getting data: { url: 'https://api.imgur.com/3/album/PAERb/images',
  headers: { Authorization: 'Client-ID 905f8623e29ae6a' } }
got data
8JQvcGS.jpg 7
Getting data: { url: 'http://i.imgur.com/8JQvcGS.jpg', encoding: 'binary' }
set imgrwal
set wallpaper
Running run number 10...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://cdna2.artstation.com/p/assets/images/images/003/452/278/large/houwai-leung-.jpg?1473823396
houwai-leung-.jpg?1473823396 13
Getting data: { url: 'https://cdna2.artstation.com/p/assets/images/images/003/452/278/large/houwai-leung-.jpg?1473823396',
  encoding: 'binary' }
set wallpaper
Running run number 11...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://cdnb1.artstation.com/p/assets/images/images/003/074/405/large/rumyana-zarkova-naga-second-panel-ver2.jpg?1469348769
rumyana-zarkova-naga-second-panel-ver2.jpg?1469348769 38
Getting data: { url: 'https://cdnb1.artstation.com/p/assets/images/images/003/074/405/large/rumyana-zarkova-naga-second-panel-ver2.jpg?1469348769',
  encoding: 'binary' }
set wallpaper
Running run number 12...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: http://imgur.com/a/swhrw
Getting data: { url: 'https://api.imgur.com/3/image/swhrw',
  headers: { Authorization: 'Client-ID 905f8623e29ae6a' } }
Getting data: { url: 'https://api.imgur.com/3/album/swhrw/images',
  headers: { Authorization: 'Client-ID 905f8623e29ae6a' } }
got data
W7dPybX.jpg 7
Getting data: { url: 'http://i.imgur.com/W7dPybX.jpg', encoding: 'binary' }
set imgrwal
set wallpaper
Running run number 13...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://cdna2.artstation.com/p/assets/images/images/003/452/278/large/houwai-leung-.jpg?1473823396
houwai-leung-.jpg?1473823396 13
Getting data: { url: 'https://cdna2.artstation.com/p/assets/images/images/003/452/278/large/houwai-leung-.jpg?1473823396',
  encoding: 'binary' }
set wallpaper
Running run number 14...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://i.redd.it/psz1nb9etdmx.jpg
psz1nb9etdmx.jpg 12
Getting data: { url: 'https://i.redd.it/psz1nb9etdmx.jpg',
  encoding: 'binary' }
set wallpaper
Running run number 15...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://i.redd.it/jc7ln195prmx.jpg
jc7ln195prmx.jpg 12
Getting data: { url: 'https://i.redd.it/jc7ln195prmx.jpg',
  encoding: 'binary' }
set wallpaper
Running run number 16...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://i.redd.it/5yw4adtd5pmx.jpg
5yw4adtd5pmx.jpg 12
Getting data: { url: 'https://i.redd.it/5yw4adtd5pmx.jpg',
  encoding: 'binary' }
set wallpaper
Running run number 17...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://i.redd.it/pd8cc1mhgimx.jpg
pd8cc1mhgimx.jpg 12
Getting data: { url: 'https://i.redd.it/pd8cc1mhgimx.jpg',
  encoding: 'binary' }
set wallpaper
Running run number 18...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://i.redd.it/5yw4adtd5pmx.jpg
5yw4adtd5pmx.jpg 12
Getting data: { url: 'https://i.redd.it/5yw4adtd5pmx.jpg',
  encoding: 'binary' }
set wallpaper
Running run number 19...
Getting data: https://reddit.com/r/wallpapers/.json
number of children: 25
Selected URL: https://i.redd.it/5yw4adtd5pmx.jpg
5yw4adtd5pmx.jpg 12
Getting data: { url: 'https://i.redd.it/5yw4adtd5pmx.jpg',
  encoding: 'binary' }
set wallpaper
